File: frontend/src/api/tasks.ts
import client from './client'
import type { Task, TaskCreate } from '../types'

export const getTasks = (developerId?: number) =>
  client.get<Task[]>('/api/tasks', {
    params: developerId ? { developer_id: developerId } : {}
  }).then(r => r.data)

export const createTask = (payload: TaskCreate) =>
  client.post<Task>('/api/tasks', payload).then(r => r.data)

export const updateTask = (id: number, payload: Partial<TaskCreate>) =>
  client.put<Task>(`/api/tasks/${id}`, payload).then(r => r.data)

export const deleteTask = (id: number) =>
  client.delete(`/api/tasks/${id}`)

File: backend/app/routers/tasks.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import List, Optional
from app.database import get_db
from app.models.task import Task
from app.schemas.task import TaskCreate, TaskUpdate, TaskResponse

router = APIRouter(prefix="/api/tasks", tags=["tasks"])

@router.get("/", response_model=List[TaskResponse])
def list_tasks(
    developer_id: Optional[int] = Query(None),
    db: Session = Depends(get_db)
):
    query = db.query(Task)
    if developer_id:
        query = query.filter(Task.developer_id == developer_id)
    return query.order_by(Task.date.desc()).all()

@router.post("/", response_model=TaskResponse, status_code=201)
def create_task(payload: TaskCreate, db: Session = Depends(get_db)):
    task = Task(**payload.model_dump())
    db.add(task)
    db.commit()
    db.refresh(task)
    return task

@router.put("/{task_id}", response_model=TaskResponse)
def update_task(task_id: int, payload: TaskUpdate, db: Session = Depends(get_db)):
    task = db.query(Task).filter(Task.id == task_id).first()
    if not task:
        raise HTTPException(status_code=404, detail="Task not found")
    for field, value in payload.model_dump(exclude_none=True).items():
        setattr(task, field, value)
    db.commit()
    db.refresh(task)
    return task

@router.delete("/{task_id}", status_code=204)
def delete_task(task_id: int, db: Session = Depends(get_db)):
    task = db.query(Task).filter(Task.id == task_id).first()
    if not task:
        raise HTTPException(status_code=404, detail="Task not found")
    db.delete(task)
    db.commit()
